### PR TITLE
[MM-26771] server: reduce code dupl. in handlers and add some error propogation

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This PR centralizes the getting `model.PullRequest` from an `IssueComment`. Also adds error returns to the handlers so that logging in the handlers can be omitted.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26771

### CODE REVIEW

File: model/pull_request.go [hunk 1]
```diff
@@ -28,4 +28,5 @@ type PullRequest struct {
 	URL                 string
 	CreatedAt           time.Time
 	MaintainerCanModify bool
+	Merged              bool
```
Comment (ethervoid) [inline model/pull_request.go:31]: Also, you need to add:

- SQL migration file [like this one](https://github.com/mattermost/mattermost-mattermod/pull/164/files#diff-9e8f48247685706a15181d89c05243beR1) that is going to be executed by @metanerd . You can use a default value [to false](https://github.com/google/go-github/blob/master/github/github-accessors.go#L8986) but I rather prefer to leave it null, use the NULLXXX field and then make the `Ticket` updates it (it's going to do it anyway) 
- Given that the flag comes from the response you should also add something like [this](https://github.com/mattermost/mattermost-mattermod/pull/164/files#diff-fec73530a7655cd7c507f18fc6ed23cdR242)

File: server/autoassign.go [hunk 2]
```diff
@@ -2,48 +2,54 @@ package server
...
 
-	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-mattermod/model"
 
 	"github.com/google/go-github/v32/github"
 )
 
-func (s *Server) handleAutoassign(ctx context.Context, eventIssueComment IssueComment) {
-	teams, _, err := s.GithubClient.Repositories.ListTeams(ctx, *eventIssueComment.Repository.Owner.Login, *eventIssueComment.Repository.Name, nil)
+func (s *Server) handleAutoassign(ctx context.Context, url string, pr *model.PullRequest) error {
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/autoassign.go]: This is to keep it consistent with others, actually we can drop `handle` prefix in all occurrences if you agree.

File: server/pull_request.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ func (s *Server) handlePullRequestEvent(ctx context.Context, event *PullRequestE
 	switch event.Action {
 	case "opened":
 		mlog.Info("PR opened", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number))
-		s.checkCLA(ctx, pr)
+		_ = s.checkCLA(ctx, pr)
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/pull_request.go]: It was being ignored before and I just wanted to make it explicitly :) We can log it anyway.

File: server/autoassign.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (agnivade) [reply server/autoassign.go]: Sorry, I meant handleAutoAssign. The second A needs to be capital :)

File: server/cla.go [hunk 4]
```diff
@@ -17,35 +18,12 @@ import (
...
-	if err != nil {
-		mlog.Error("pr_error", mlog.Err(err))
-		return
-	}
-
-	if *prGitHub.State == model.StateClosed {
-		return
+func (s *Server) handleCheckCLA(ctx context.Context, pr *model.PullRequest) error {
+	if pr.State == model.StateClosed {
+		return errors.New("pull request is closed")
```
Comment (agnivade) [inline server/cla.go:23]:
```suggestion
		return nil
```

**Pull request merged**